### DISCUSSION TITLE
nextcloud collabora: Translation of languages takes ages to load

### DISCUSSION THREAD

**Discussion by leon1995** [ZoeyVid/NPMplus#1127] [Q&A] [answered]
Hi,
I created a [ticket in the nextcloud forum](https://help.nextcloud.com/t/translation-of-languages-takes-ages-to-load/205128) describing problems when using collabora with npmplus. I tried to use the "normal" nginx proxy manager and the problem disappeared.
Please note that I am not using the community container but the standalone one on a different host

**Comment by Zoey2936** [comment]
Do you use crowdsec or modsec?

**Comment by leon1995** [comment]
crowdsec yes, modsec no

EDIT: configuration
![Screenshot 2024-09-30 at 10 42 50](https://github.com/user-attachments/assets/579da7a4-51cf-416d-8c33-ea1e0cca92ec)
![Screenshot 2024-09-30 at 10 43 45](https://github.com/user-attachments/assets/acf01c70-3cbe-4758-ac49-8d2bf52f7a65)
![Screenshot 2024-09-30 at 10 43 59](https://github.com/user-attachments/assets/0bc2a527-3b4d-47b2-a456-51ac6199cd91)

**Comment by Zoey2936** [comment]
Can you maybe try with crowdsec off?

**Comment by leon1995** [comment]
starting the npmplus without crowdsec does not work as I get blocked by crowdsec? How do I turn it off? Do I have to create a completly new setup?

**Comment by Zoey2936** [comment]
You open the crowdsec config file change ENABLED to false and restart the container

**Comment by leon1995** [comment]
I disabled it and restarted both container but didnt helped. Same result
EDIT: here is a screenshot form the firefox network tab showing the problematic urls
![Screenshot 2024-09-30 at 10 59 14](https://github.com/user-attachments/assets/09b7b87d-9f15-47c8-9327-cb51d442889d)

**Comment by Zoey2936** [comment]
Is there something written in npmplus docker logs?

**Comment by leon1995** [comment]
not that much, just
```
npmplus  | 2024/09/30 11:11:14 [error] 1050#1050: *1460 upstream prematurely closed connection while reading upstream, client: redacted_client_ip, server: nextcloud.mydomain, request: "GET /browser/d5ebff5/l10n/help-de.json HTTP/2.0", upstream: "http://redacted_server_ip/browser/d5ebff5/l10n/help-de.json", host: "nextcloud.mydomain"
npmplus  | 2024/09/30 11:12:14 [error] 1050#1050: *1460 upstream prematurely closed connection while reading upstream, client: redacted_client_ip, server: nextcloud.mydomain, request: "GET /browser/d5ebff5/l10n/uno/de-override.json HTTP/2.0", upstream: "http://redacted_server_ip/browser/d5ebff5/l10n/uno/de-override.json", host: "nextcloud.mydomain"
```

**Comment by Zoey2936** [comment]
Can you disable proxy buffering in your npmplus compose file please?

**Comment by leon1995** [comment]
I commented in `"NGINX_DISABLE_PROXY_BUFFERING=true"` and tried with and without crowdsec. Didnt helped

**Comment by Zoey2936** [comment]
proxy_read_timeout 300s; proxy_connect_timeout 75s;

**Comment by Zoey2936** [comment]
Can you add this to your advanced config?

**Comment by leon1995** [comment]
I added it to the advanced config, but didnt helped
```
npmplus  | [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf
npmplus  | [Nginx    ] › ℹ  info      Reloading Nginx
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [warn] 1003#1003: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
npmplus  | 2024/09/30 11:37:54 [alert] 1859#1859: sched_setaffinity() failed (22: Invalid argument)
npmplus  | 2024/09/30 11:38:18 [error] 1859#1859: *1285 upstream prematurely closed connection while reading upstream, client: redacted_client_ip, server: nextcloud.mydomain, request: "GET /browser/d5ebff5/l10n/help-de.json HTTP/2.0", upstream: "http://redacted_server_ip/browser/d5ebff5/l10n/help-de.json", host: "nextcloud.mydomain"
npmplus  | 2024/09/30 11:39:19 [error] 1859#1859: *1285 upstream prematurely closed connection while reading upstream, client: redacted_client_ip, server: nextcloud.mydomain, request: "GET /browser/d5ebff5/l10n/uno/de-override.json HTTP/2.0", upstream: "http://redacted_server_ip/browser/d5ebff5/l10n/uno/de-override.json", host: "nextcloud.mydomain"
```

**Comment by Zoey2936** [comment]
I can reproduce it:
```
root@mx:~# curl http://127.0.0.1:11000/browser/d5ebff5/l10n/help-de.json 
curl: (18) transfer closed with outstanding read data remaining
{}
```

**Comment by Zoey2936** [comment]
since even curl throws an error, I have no idea what the problem could be

**Comment by leon1995** [comment]
It has to be related to the changes you made, because its working with the base npm.
Its good that you were able to reproduce it.
What can I do to help?

**Comment by Zoey2936** [comment]
NPMplus makes all request "uncompressed" in diffrence to NPM, so it can compress itself: https://github.com/ZoeyVid/NPMplus/blob/8dfb3d27ec26a91dbfe4ac2ae543eaf73ed5ef1f/rootfs/usr/local/nginx/conf/conf.d/include/proxy-location.conf#L6, if I make the curl request compressed, it works. So this would be a workarround (NPMplus requests compressed to upstream). BUT the request to colabora should also work uncomnpressed

**Comment by Zoey2936** [comment]
I could fix it, but I see the problem with collabora

**Comment by leon1995** [comment]
Maybe we should create an issue in collabora? Maybe they know more

**Comment by Zoey2936** [comment]
The problem seems to be caddy:
```
root@mx:~# curl http://172.20.0.6:9980/browser/d5ebff5/l10n/help-de.json; echo $? # ip of nextcloud-aio-collabora
{}0

root@mx:~# curl http://172.20.0.18:11000/browser/d5ebff5/l10n/help-de.json; echo $? # ip of nextcloud-aio-apache
curl: (18) transfer closed with outstanding read data remaining
{}18
```

**Comment by leon1995** [comment]
Where is caddy used? I thought nextcloud used apache? Or is collabora using caddy?

**Comment by Zoey2936** [comment]
https://github.com/nextcloud/all-in-one/blob/c993a66eafe495bab6947c2da3815c13be720c41/Containers/apache/Caddyfile#L23

**Comment by Zoey2936** [comment]
so this needs to be reported to Caddy (or AIO)

**Comment by leon1995** [comment]
I can report this to aio tomorrow. However, I do not have enough technical knowledge to report this to caddy 😅

**Reply by leon1995** [reply]
I created an issue https://github.com/nextcloud/all-in-one/issues/5352

**Comment by Zoey2936** [comment]
does the issue still exist for you with NPmplus?

**Reply by Zoey2936** [reply]
since I enabled the upstream compression

**Reply by leon1995** [reply]
I switched back to the original npm and havent tried since. Ill check it in the following days. Do I just have to pull the image?

**Reply by Zoey2936** [reply]
yes

**Reply by leon1995** [reply]
seems to work now with brotli, modsec and crowdsec :)

**Comment by Joly0** [comment]
Hey, should this still be an issue? Because i just noticed, that i am affected aswell:
```
2025/03/03 11:01:08 [error] 1790#1790: *1246 upstream prematurely closed connection while reading upstream, client: myIP, server: nextcloud.myDomain.com, request: "GET /browser/321ff0a561/l10n/uno/de-override.json HTTP/2.0", upstream: "http://192.168.178.220:11000/browser/321ff0a561/l10n/uno/de-override.json", host: "nextcloud.myDomain.com"
```
i have brotli enabled, but modsec and crowdsec disabled

**Reply by Zoey2936** [reply, marked as answer]
can you create a custom location with the path `/`, con figure it to proxy to aio, exactly like you also did in the details tab, then press the gear and then paste this in the box which is then shown and save: `proxy_set_header Accept-Encoding $http_accept_encoding;`

**Reply by Zoey2936** [reply]
Please note that this is only a workaround and the issues seems to be collabora and/or caddy

**Reply by Joly0** [reply]
Ok, that solved the issue. I also commented under the open issue on the collabora repo, but it looks like there is no progress about this at all... sadly...
But glad i have a workaround for now. Thank you :)
